traffic_ops/testing/compare/compare.go

```diff
@@ -157,74 +159,301 @@ func testRoute(tos []*Connect, route string) {
+					if line[:len(LONG_HEADER)] != LONG_HEADER {
+						scrubbedResult1 += line
 					}
 				} else {
-					testResult += testLine
+					scrubbedResult1 += line
 				}
 			}
 		}
+	} else {
+		scrubbedResult0 = result0Str
+		scrubbedResult1 = result1Str
 	}
 
-	if refResult == testResult {
-		log.Printf("Identical results (%d bytes) from %s", len(res[0].Res), route)
+	if scrubbedResult0 == scrubbedResult1 {
+		log.Printf("Identical results (%d bytes) from %s\n", len(result0), route)
 	} else {
-		log.Print("Diffs from ", route, " written to")
-		for _, r := range res {
-			p, err := r.TO.writeResults(route, r.Res)
-			if err != nil {
-				log.Fatalf("Error writing results for %s: %s", route, err.Error())
+		writeAllResults(route, result0Str, (*responses)[0].TO, result1Str, (*responses)[1].TO)
+	}
+}
+
+// Given a slice of (exactly two) result objects, compares the JSON content of their responses
+// and write them to files if they differ. Ignores timestamps and Traffic Ops hostnames (to the
+// degree possible)
+func handleJSONResponse(responses *[]result, route string) {
+
+	// I avoid using `defer` to close the bodies because I want to do it as quickly as possible
+
+	result0Orig, err := ioutil.ReadAll((*responses)[0].Res.Body)
+	(*responses)[0].Res.Body.Close()
+	if err != nil {
+		(*responses)[1].Res.Body.Close()
+		log.Fatalf("Failed to read response body from %s/%s: %s\n", (*responses)[0].TO.URL, route, err.Error())
+	}
+	var result0 interface{}
+	if err = json.Unmarshal([]byte(result0Orig), &result0); err != nil {
+		log.Fatalf("Failed to parse response body from %s/%s as JSON: %s\n", (*responses)[0].TO.URL, route, err.Error())
+	}
+
+	result1Orig, err := ioutil.ReadAll((*responses)[1].Res.Body)
+	(*responses)[1].Res.Body.Close()
+	if err != nil {
+		log.Fatalf("Failed to read response body from %s/%s: %s\n", (*responses)[1].TO.URL, route, err.Error())
+	}
+	var result1 interface{}
+	if err = json.Unmarshal(result1Orig, &result1); err != nil {
+		log.Fatalf("Failed to parse response body from %s/%s as JSON: %s\n", (*responses)[1].TO.URL, route, err.Error())
+	}
+
+	// type-safety cast to maps
+	m0, m1 := result0.(map[string]interface{}), result1.(map[string]interface{})
+
+	// ... Now we have to iterate over every key in each map to determine if it should be removed
+	for key, value := range m0 {
+
+		// handles snapshot/new timestamps/hostnames/versions
+		if (key == "response") {
+			switch Type := value.(type) {
+				case map[string]interface{}:
+					if k, in := value.(map[string]interface{})["stats"]; in {
+						switch t := k.(type) {
+							case map[string]interface{}:
+								if val, ok := k.(map[string]interface{})["date"]; ok {
+									switch T := val.(type) {
+										case float64:
+											delete(result0.(map[string]interface{})["response"].(map[string]interface{})["stats"].(map[string]interface{}), "date")
+										default:
+											log.Printf("found 'stats.date' object - wasn't an integer - %v", T)
```

Comment:
If it's not a string, then it's not the field I'm looking for and that would mean I'm deleting an object that _should_ be included in the comparison. Doing this generically just isn't possible, unfortunately, since at its core this is a hard-coded handling of specific keys for specific endpoints.

The other possibility I looked at was doing if statements looking at the actual route name, but ultimately I decided that this would be more forward-compatible with changes to the names of routes and introduction of new routes.

What I definitely can (and should) do is move the logic for doing this modification to a JSON object into its own `func` and just call that on each of the responses.

Reply:
FYI I also figured out that Go `switch`s don't require assigning lvalues to the output, which let me remove all of those `default` cases that I just had in there because I thought it was required...